File: Sarvam C-DAC Project/Sarvam C-DAC Spring Boot/Sarvam/src/main/java/com/cdac/sarvam/model/MasterModuleSubpoint.java
package com.cdac.sarvam.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "master_module_subpoints")
public class MasterModuleSubpoint {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer subId;

    @ManyToOne
    @JoinColumn(name = "module_id", nullable = false)
    private MasterModule module;

    @Column(name = "module_subpoint_name", length = 100, nullable = false)
    private String moduleSubpointName;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();
}

File: MS.NET/Sarvam Backend/Models/MasterModuleSubpoint.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Sarvam.Models
{
    public class MasterModuleSubpoint
    {
        public int SubId { get; set; } // Primary Key (PK)
        public int ModuleId { get; set; } // Foreign Key to MasterModule
        public string ModuleSubpointName { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;

        // Navigation Property
        [NotMapped]
        public MasterModule MasterModule { get; set; } // Reference to the MasterModule
        [NotMapped]
        public MasterProfile MasterProfile { get; set; }
    }
}
